src/maus/mig_ahb_matching.py

```diff
@@ -0,0 +1,216 @@
+                    except ValueError:
+                        pass
+        if any((True for line in data_element_lines if line.data_element is not None)):
+            if not any((True for line in data_element_lines if line.ahb_expression is not None)):
+                # if none of the items is marked with an ahb expression it's probably not required in this AHB
+                continue
+            data_element = merge_lines_with_same_data_element(data_element_lines, first_stack=stack)
+            # try:
+            append_next_data_elements_here.append(data_element)  # pylint:disable=used-before-assignment
+            # except UnboundLocalError as unbound_local_error:
+            #    raise ValueError(f"No segment has been created for {stack}") from unbound_local_error
+        else:
+            first_line = first(data_element_lines)
+            last_line = last(data_element_lines)
+            if (
+                first_line.segment_group_key == last(position.layers).segment_group_key
+                and last(position.layers).opening_segment_code == last_line.segment_code
+                and stack is not None
+                and stack.to_json_path() not in used_stacks
+                and first_line.ahb_expression is not None
+            ):
+                # a new segment group has been opened
+                segment_group = SegmentGroup(
+                    discriminator=stack.to_json_path(),
+                    # type:ignore[arg-type] # might be None now, will be replaced later
+                    ahb_expression=first_line.ahb_expression.strip(),
+                    segments=[],
+                    segment_groups=[],
+                    ahb_line_index=first_line.index,
+                )
+                used_stacks.add(stack.to_json_path())
+                append_next_segments_here = segment_group.segments  # type:ignore[assignment]
+                if segment_group.discriminator == '$["Dokument"][0]["Nachricht"][0]':
+                    result.lines.append(segment_group)
+                    append_next_sg_here = segment_group.segment_groups  # type:ignore[assignment]
+                    parent_group_lists.append(result.lines)
+                elif position.is_sub_location_of(previous_position):  # pylint:disable=used-before-assignment
+                    append_next_sg_here.append(segment_group)
+                    parent_group_lists.append(append_next_sg_here)
+                    append_next_sg_here = segment_group.segment_groups  # type:ignore[assignment]
+                else:
+                    distance = calculate_distance(previous_position, position)
+                    for _ in range(0, distance.layers_up):
+                        # try:
+                        append_next_sg_here = parent_group_lists.pop()
+                        # except IndexError as index_error:
+                        #    raise ValueError(f"Couldn't move from {previous_position} to {position}") from index_error
```

Comment:
braucht es die auskommentierten Zeilen noch?

Reply:
ja, sie helfen, wenn es bricht. hatte sie auskommentiert um die anzahl branches in der methode zu verkleinern aber das ist gerade eh hoffnugnslos: 1c3046ddeecd9c3364890fd15e3000fcd1f3ce60